lib/_charnames.pm

```diff
@@ -785,12 +785,12 @@ sub viacode {
     # Return the official name, if exists.  It's unclear to me (khw) at
     # this juncture if it is better to return a user-defined override, so
     # leaving it as is for now.
-    if ($txt =~ m/^$hex\t/m) {
+    if ($txt =~ m/^$hex$/m) {
 
         # The name starts with the next character and goes up to the
         # next new-line.  Using capturing parentheses above instead of
         # @+ more than doubles the execution time in Perl 5.13
-        $return = substr($txt, $+[0], index($txt, "\n", $+[0]) - $+[0]);
+        $return = substr($txt, $+[0] + 1, index($txt, "\n", $+[0] + 1) - ($+[0] + 1));
```

Comment:
If you change the pattern to `m/^$hex\n/m`, you don't have to mess with all the +1s.

Reply:
done

Reply:
done